akka-persistence/src/test/scala/akka/persistence/SnapshotFailureRobustnessSpec.scala

```diff
@@ -30,6 +36,26 @@ object SnapshotFailureRobustnessSpec {
     }
   }
 
+  class DeleteSnapshotTestPersistentActor(name: String, probe: ActorRef) extends NamedPersistentActor(name) {
+
+    // TODO do we call it "snapshot store" or "snapshot plugin", small inconsistency here
+    override def snapshotPluginId: String =
+      "akka.persistence.snapshot-store.local-delete-fail"
+
+    override def receiveRecover: Receive = {
+      case SnapshotOffer(md, s) ⇒ probe ! ((md, s))
+      case other                ⇒ probe ! other
+    }
+
+    override def receiveCommand = {
+      case Cmd(payload)            ⇒ persist(payload)(_ ⇒ saveSnapshot(payload))
+      case DeleteSnapshot(seqNr)   ⇒ deleteSnapshot(seqNr, (new Date).getTime) // TODO shouldn't timestamp be optional?
```

Comment:
Ah now I see... it's reusing the metadata type that is stored with snapshots. Let me re think this